sw/host/opentitanlib/src/util/usr_access.rs

```diff
@@ -35,3 +66,34 @@ pub fn usr_access_get(bitstream: &[u8]) -> Result<u32> {
     Ok(usr_access)
 }
 
+/// Returns a 32-bit timestamp suitable to be used as a USR_ACCESS value:
+///
+/// |-------+-------+-------+-------------------------+-------+--------+--------|
+/// | Bits: | 31:27 | 26:23 | 22:17                   | 16:12 | 11:6   | 5:0    |
+/// |-------+-------+-------+-------------------------+-------+--------+--------|
+/// | Data: | Day   | Month | Year (since 2000, 0-63) | Hour  | Minute | Second |
+/// |-------+-------+-------+-------------------------+-------+--------+--------|
+///
+/// From
+/// https://www.xilinx.com/content/dam/xilinx/support/documents/application_notes/xapp1232-bitstream-id-with-usr_access.pdf
+pub fn usr_access_timestamp() -> u32 {
+    let now = Utc::now();
+    now.day() << 27
+        | now.month() << 23
+        | now.year_ce().1.checked_sub(2000u32).unwrap() << 17
+        | now.hour() << 12
+        | now.minute() << 6
+        | now.second()
+}
+
+pub fn usr_access_set(bitstream: &mut [u8], val: u32) -> Result<()> {
```

Comment:
WDYT about adding some tests for `usr_access_set()`? I'd prioritize testing the new code, so mainly `remove_crc()`, I suppose.

Reply:
I agree but I'd rather do this later since we have more important tests that we should prioritize, tracked in #14880.